src/com/superliminal/magiccube4d/MC4DSwing.java

```diff
@@ -1091,6 +1094,17 @@ public void actionPerformed(ActionEvent ae) {
         }
     } // initPuzzleMenu
 
+    // Quick hack for figuring out what puzzleManager.puzzleDescription.getIntLength()
+    // is going to be, before we can query it (that is, while puzzleManager.puzzleDescription
+    // is being computed in the background).
+    // That is, just peel off an int from the beginning of the length string.
+    // E.g. "4"->4,  "3(9.0)"->3.
+    // Note that this will throw if there is no int at the beginning.
+    private static int predictIntLengthFromLengthString(String lengthString)
```

Comment:
Presumably, this prediction might not match (in all cases) what the puzzle description might end up returning. I wondered what the consequences of that may be.

Reply:
This is certainly a shaky hack.   It relies on the contract (that I probably haven't even written down anywhere) that the lengthString is of exactly this form: it is, precisely, a nonnegative int optionally followed by a parenthesized double.

That's true at this moment, but it's fragile-- in fact, the upcoming version of PolytopePuzzleDescription is more general in that it allows multiple lengths, one for each edge direction.  E.g. in the dev app in the donhatchsw repo, "(4)x(3)x(2)x(1) 4,3,2,1" is a nonuniform box puzzle:
![nonuniform_box](https://user-images.githubusercontent.com/12412901/92323868-f16f1580-eff0-11ea-90d4-1bf5f885369d.png)

OTOH, with that comes the fact that puzzles won't even have a single intLength and doubleLength that can be queried any more.  So this function is part of a bunch of stuff that will have to be reworked if I port to that new version (and assuming I keep the ability to do nonuniform boxes like this, which is itself kind of a special-purpose hack).